### PR TITLE
VideoPlayerVideo: Need to clear pictures before switching decoder

### PR DESCRIPTION
**Pull request by popcornmix**
The new decoder may not know how clear a picture initialised from a different decoder.
For example MMALCodec wants Release() called on clear, but VideoCodecFFMPEG does not.

### CODE REVIEW

**Comment by popcornmix** [comment]
ping @FernetMenta

File: xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp [hunk 1]
```diff
@@ -179,8 +179,11 @@ void CVideoPlayerVideo::OpenStream(CDVDStreamInfo &hint, CDVDVideoCodec* codec)
     m_fForcedAspectRatio = 0.0;
 
   if (m_pVideoCodec)
+  {
+    m_pVideoCodec->ClearPicture(&m_picture);
+    m_pVideoCodec->Dispose();
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp]: please call Dispose() in destructor

other changes look ok

**Comment by popcornmix** [comment]
@FernetMenta updated

File: xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp [hunk 2]
```diff
@@ -99,6 +99,8 @@ CVideoPlayerVideo::CVideoPlayerVideo(CDVDClock* pClock
 
 CVideoPlayerVideo::~CVideoPlayerVideo()
 {
+  if (m_pVideoCodec)
+    m_pVideoCodec->Dispose();
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp]: Sorry, I meant the destructor of the codec. Then we can drop Dispose() from the interface of VideoCodec.

**Comment by popcornmix** [comment]
@FernetMenta updated

**Comment by FernetMenta** [comment]
+1, thanks

**Comment by popcornmix** [comment]
jenkins build this please

**Pull request merged**